lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -180,6 +182,60 @@ def initconfig_compiler_entries(self):
 
         return entries
 
+    class Scheduler(enum.Enum):
+        LSF = enum.auto()
+        SLURM = enum.auto()
+        FLUX = enum.auto()
+
+    def get_scheduler(self) -> Optional[Scheduler]:
+        spec = self.pkg.spec
+        if spec.satisfies("^spectrum-mpi") or spec["mpi"].satisfies("schedulers=lsf"):
```

Comment:
This could probably use a comment or two to clarify/document. Like spectrum-mpi always uses lsf. The TODO is very helpful.

Reply:
Added some comments in dcf19395faa